Sources/XiEditor/XiDocument.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+
+protocol XiDocument: class {
+    func resize(size: CGSize)
+}
+
+class XiDocumentConnection: XiDocument {
+
+    typealias AsyncRpc = (_ method: String, _ params: Any, _ callback: RpcCallback?) -> Void
+    typealias SyncRpc = (_ method: String, _ params: Any) -> RpcResult
+
+    private let asyncRpc: AsyncRpc
+    private let syncRpc: SyncRpc
+
+    init(asyncRpc: @escaping AsyncRpc, syncRpc: @escaping SyncRpc) {
```

Comment:
the idea with passing these in as closures is to make this easier to mock? Or this is just kind of the 'swifty' way? I guess historically i would expect some delegate protocol or something but maybe we're moving away from that?

Reply:
This is just the initial idea of incorporating the `XiDocumentConnection` into the existing code. With this approach we'll be able to pass in [`sendRpcAsync`](https://github.com/xi-editor/xi-mac/blob/master/Sources/XiEditor/Document.swift#L118) and [`sendRpc`](https://github.com/xi-editor/xi-mac/blob/master/Sources/XiEditor/Document.swift#L131) from existing [`Document`](https://github.com/xi-editor/xi-mac/blob/master/Sources/XiEditor/Document.swift#L23) class and have existing `viewIdentifier`, tracing and queueing existing notifications for free 👯. Once we have that we'll be able to iterate on the code and decouple it 😄